pkg/cluster/builder.go

```diff
@@ -125,7 +126,9 @@ func (b FileSpecBuilder) Build(clusterConfigURL string) (*Spec, error) {
 	}
 
 	releaseVersion := v1alpha1.EksaVersion(release.Version)
-	config.Cluster.Spec.EksaVersion = &releaseVersion
+	if config.Cluster.IsSelfManaged() || (config.Cluster.IsManaged() && !features.OverrideEksaVersion().IsActive()) {
```

Comment:
why are we using a feature flag here? I probably didn't understand what we are trying to do here

Reply:
We don't need the feature flag. I was just using it to test what would happen if we let the user pass an eksaVersion value where it was previously set by the CLI and vice versa.